typeset/converter.js

```diff
@@ -129,73 +146,44 @@ const createMapOfMathMLElements = async (log, inputPath, cssPath, outputPath, ou
       index++
     }
     return [...mathMap.entries()]
-  }, PROGRESS_TIME)
-
-  const convertedMathML/*: Map<string, {svg, html, css}> */ = await mjnodeConverter.convertMathML(log, new Map(mathEntries), outputFormat)
+  }, PROGRESS_TIME, mathNodePlaceholder('{id}'))
```

Comment:
Yes, because I couldn't pass `mathNodePlaceholder` as a function to `page.evaluate`, what I decided to do is to create something like `<!-- math node {id} -->` and then inside `page.evaluate`, do `PLACE_HOLDER.replace('{id}', id)`.  So it is still outputting the right thing.  If you have a better solution, I'll be very happy to change it though.

Reply:
What about changing this part:
```
// only set an ID if one does not already exist
      if (!mathNode.getAttribute('id')) {
        mathNode.setAttribute('id', `mjnode-${index}`)
        mathNode.classList.add('-remove-id-later')
      }

      const id = mathNode.getAttribute('id')
      // put html comment placeholder for where the converted math should go
      mathNode.outerHTML = PLACE_HOLDER.replace('{id}', id)
```
to something like:
```
const id = mathNode.getAttribute('id') || `mjnode-${index}`
mathNode.outerHTML = PLACE_HOLDER.replace('{id}', id)
```
where `PLACE_HOLDER` is just `const PLACE_HOLDER = "<!-- math node {id} -->"`

Reply:
I don't want to have multiple places that defines that constant `<!-- math node ${id} -->`...  I can rename it to `PLACE_HOLDER_TEMPLATE`?